File: The_Cryptopals_Crypto_Challenges/Set1/Challenge_7.go
package main

import (
	"crypto/aes"
	"encoding/base64"
	"fmt"
	"os"
)

func AES_ECB_Decrypt(key, data []byte) (string, error) {
 ciph, err := aes.NewCipher(key)
 if err != nil {
   return "",err
 }
 blockSize := ciph.BlockSize()
 plaintext := make([]byte,len(data))
 for i,j := 0,blockSize; i < len(data); i,j = i+blockSize,j+blockSize {
   ciph.Decrypt(plaintext[i:j],data[i:j])
 }
 return string(plaintext),nil
}

func main() {
  b64_data,err := os.ReadFile("7.txt")
  if err != nil {
    fmt.Println(err)
    return
  }
  raw_data := make([]byte,len(b64_data))
  l,err := base64.RawStdEncoding.Decode(raw_data,b64_data)
  if err != nil {
    fmt.Println(err)
    return
  }
  key := []byte("YELLOW SUBMARINE")
  plaintext, err := AES_ECB_Decrypt(key,raw_data[:l])
  if err != nil {
    fmt.Println(err)
    return
  }
  fmt.Println(plaintext)
}

File: The_Cryptopals_Crypto_Challenges/Set1/Challenge_7.py
from Crypto.Cipher import AES
import base64

def AES_ECB_Decrypt(key, data):
    cipher = AES.new(key,AES.MODE_ECB)
    plaintext = cipher.decrypt(data)
    return plaintext


try:
    with open('7.txt','r') as file:
        data = file.read()
        raw_data = base64.b64decode(data)
        key = b"YELLOW SUBMARINE"
        plaintext = AES_ECB_Decrypt(key,raw_data)
        print(plaintext.decode())

except FileNotFoundError:
    print("File not found")
